provider/rfc2136/rfc2136.go

```diff
@@ -439,3 +456,14 @@ func chunkBy(slice []*endpoint.Endpoint, chunkSize int) [][]*endpoint.Endpoint {
 
 	return chunks
 }
+
+func findMsgZone(ep *endpoint.Endpoint, m *dns.Msg, r rfc2136Provider) string {
```

Comment:
With this clearer method, I suggest:
1. To remove dns.Msg since it's not used in the function
2. To pass `r.zoneNames` instead of a copy of `rfc2136Provider`

Reply:
fixed: https://github.com/kubernetes-sigs/external-dns/pull/3976/commits/411acc78950e8647fcb5a453549995f71a0775fd